### PR TITLE
Add wrapper for hb_blob_t

### PR DESCRIPTION
**Pull request by mbrubeck**
This change is [Reviewable](https://reviewable.io/reviews/servo/rust-harfbuzz/109)

### CODE REVIEW

File: harfbuzz/src/blob.rs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        unsafe { sys::hb_blob_get_length(self.raw) as usize }
+    }
+
+    /// Make this blob immutable.
+    pub fn make_immutable(&mut self) {
+        unsafe { sys::hb_blob_make_immutable(self.raw); }
+    }
+
+    /// Returns true if the blob is immutable.
+    pub fn is_immutable(&mut self) {
```
Comment (waywardmonkeys) [inline harfbuzz/src/blob.rs]: This looks like a copy paste error ... it is the same as the above function.

File: harfbuzz/src/blob.rs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+    /// Construct a `Blob` from a raw pointer. Takes ownership of the blob.
+    pub unsafe fn from_raw(raw: *mut sys::hb_blob_t) -> Self {
+        Blob { raw }
+    }
+
+    /// Returns the size of the blob in bytes.
+    pub fn len(&self) -> usize {
+        unsafe { sys::hb_blob_get_length(self.raw) as usize }
+    }
```
Comment (waywardmonkeys) [inline harfbuzz/src/blob.rs]: Suspect that with a `len` function, clippy will warm if there isn’t an `is_empty` function.

File: harfbuzz/src/blob.rs (hunk 1 above)
Comment (mbrubeck) [reply harfbuzz/src/blob.rs]: Oops! Fixed.

File: harfbuzz/src/blob.rs (hunk 2 above)
Comment (mbrubeck) [reply harfbuzz/src/blob.rs]: Fixed.

**Pull request merged**